cranelift/codegen/src/timing.rs

```diff
@@ -2,12 +2,13 @@
 //!
 //! This modules provides facilities for timing the execution of individual compilation passes.
 
-use core::fmt;
-use std::any::Any;
-use std::boxed::Box;
-use std::cell::RefCell;
-use std::mem;
-use std::time::Duration;
+use alloc::boxed::Box;
+use alloc::fmt;
+use core::any::Any;
+use core::time::Duration;
+
+#[cfg(feature = "timing")]
+use core::cell::RefCell;
```

Comment:
What I might recommend for this file is to have a `timing.rs` and a `timing_disabled.rs` as opposed to a single source file with lots of `#[cfg]`, that's worked pretty well for us in the past.